jobs/V3/basic_job_sample.rb

```diff
@@ -12,213 +12,228 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-require "google/apis/jobs_v3"
-require "rails"
-require 'securerandom'
-require_relative 'basic_company_sample'
-
-=begin
-	This file contains the basic knowledge about company and job, including:
-	- Construct a company with required fields
-	- Create a company
-	- Get a company
-	- Update a company
-	- Update a company with field mask
-	- Delete a company
-=end
-
-class BasicJobSample
-
-	# Instantiate the client
-	@@Jobs   = Google::Apis::JobsV3
-	# ProjectId to get company list
-	@@DEFAULT_PROJECT_ID = "projects/" + ENV["GOOGLE_CLOUD_PROJECT"];
-
-
-	@@talentSolution_client = @@Jobs::CloudTalentSolutionService.new
-	# @see https://developers.google.com/identity/protocols/application-default-credentials#callingruby
-	@@talentSolution_client.authorization = Google::Auth.get_application_default(
-		"https://www.googleapis.com/auth/jobs"
+def generate_job(company_name)
+	# [START generate_job]
+	require "google/apis/jobs_v3"
+	require "securerandom"
+
+	jobs   = Google::Apis::JobsV3;
+
+	requisition_id = "jobWithRequiredFields:" + SecureRandom.hex;
+	application_info = jobs::ApplicationInfo.new;
+	application_info.uris = Array["http://careers.google.com"];
+	job_generated = jobs::Job.new;
+	job_generated.requisition_id = requisition_id;
+	job_generated.title = " Lab Technician";
+	job_generated.company_name = company_name;
+	job_generated.employment_types = Array["FULL_TIME"];
+	job_generated.language_code = "en-US";
+	job_generated.application_info = application_info;
+	job_generated.description = "Design, develop, test, deploy, maintain and improve software.";
+	
+	# set compensation to 12 USD/hour
+	compensation_info = jobs::CompensationInfo.new;
+	compensation_entry = jobs::CompensationEntry.new;
+	compensation_amount = jobs::Money.new;
```

Comment:
Could you try setting the value for compensation_amount using the following?:
```ruby
{
   currency_code: "USD",
   units: 12,
}
```

If it works for you apply it to other object representations. It's hard to parse each instance representing parts of the request.

Reply:
I made the changes based on my understand. Pls check whether it's as expected. Thanks! :)